compilers/concrete-optimizer/concrete-optimizer/src/optimization/dag/multi_parameters/analyze.rs

```diff
@@ -61,10 +62,34 @@ pub fn analyze(
         Some(p_cut) => p_cut.clone(),
         None => maximal_p_cut(&dag),
     };
-    let partitions = partitionning_with_preferred(&dag, &p_cut, default_partition);
+    let partitions =
+        partitionning_with_preferred(&dag, &p_cut, default_partition, allow_composition);
     let instrs_partition = partitions.instrs_partition;
     let nb_partitions = partitions.nb_partitions;
-    let out_variances = out_variances(&dag, nb_partitions, &instrs_partition);
+    let mut out_variances = self::out_variances(&dag, nb_partitions, &instrs_partition, &None);
+    if allow_composition {
+        // Verify that there is no input symbol in the symbolic variances of the outputs.
+        assert_no_input_var_in_out_var(&dag, &out_variances, nb_partitions);
+        // Get the largest output out_variance
+        let largest_output_variances = dag
```

Comment:
To put in a function (def conservative_final_variance),
And a for loop here would be a lot better. The max-like operation could be in symbolic_variance. 
```rust
let mut conservative_final_variance = SymbolicVaraince::ZERO;
for (_, _, final_variance)  in extra_final_variances(...) {
    conservative_final_variance = conservative_final_variance.component_wise_max(final_variance);
} 
```

Reply:
In fact this max already exists.
https://github.com/zama-ai/concrete/blob/main/compilers/concrete-optimizer/concrete-optimizer/src/optimization/dag/multi_parameters/symbolic_variance.rs#L170